In [123]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

warnings.filterwarnings('ignore')
np.random.seed(42)
pd.set_option("display.max_rows", None)
pd.set_option("display.max_columns", None)
pd.options.display.float_format = '{:,.2f}'.format

# Trabalhando os Dados Iniciais

In [124]:
df = pd.read_excel('./ExportacaoVinhos.xlsx')
df_inicial = df[df.columns[0:31]]
df_inicial.head()

Pais  2008_KG  2008_USD   2009_KG  2009_USD  2010_KG  2010_USD  \
0          Rússia  6207658   2352768  21912914   5732280        0         0   
1        Paraguai  2191901   1374088    486927    392087   510989    449197   
2  Estados Unidos   443895    804607    372319    660066   228968    478630   
3           China     8689     25926   1553416    482400      795      2358   
4         Espanha     2942      6834      2181      4050        0         0   

   2011_KG  2011_USD  2012_KG  2012_USD  2013_KG  2013_USD  2014_KG  2014_USD  \
0        0         0  4528176   2103968  5893291  14795694   190656     61440   
1   240168    276281   354824    428279   481564    680828   521847    908028   
2   306787   1030254   146585    303986   245368    786556   222267    494216   
3    54156    334867    87905    642177    40929    279956    64040    455340   
4     5206     24618        0         0  1972980   3748940        0         0   

   2015_KG  2015_USD  2016_KG  2016_USD  2017_KG  2017_USD  2018_KG  2018_USD  \
0    47664     18240        0         0        0         0        0         0   
1   495428    741370   985739   1655417  2393468   4274650  3234168   5494321   
2   195896    524109   258072    687411   132688   1523699   169109    512519   
3    47609    222866   134106    499622    67594    266086    30835    126336   
4        0         0        0         0        0         0     6123     22631   

   2019_KG  2019_USD  2020_KG  2020_USD  2021_KG  2021_USD  2022_KG  2022_USD  
0        0         0     1463      8550   181931    312926    66046    118618  
1  2419537   3826587  3299013   3869243  6522527   7192362  5076670   7156293  
2   209765    616274   300178    610793   111085    203554   220373    447893  
3   129852    376828   122253    363000    61884    264116   105395    404647  
4     3540      1353       28       126        0         0        0         0

In [128]:
df.head()

Pais  2008_KG  2008_USD   2009_KG  2009_USD  2010_KG  2010_USD  \
0          Rússia  6207658   2352768  21912914   5732280        0         0   
1        Paraguai  2191901   1374088    486927    392087   510989    449197   
2  Estados Unidos   443895    804607    372319    660066   228968    478630   
3           China     8689     25926   1553416    482400      795      2358   
4         Espanha     2942      6834      2181      4050        0         0   

   2011_KG  2011_USD  2012_KG  2012_USD  2013_KG  2013_USD  2014_KG  2014_USD  \
0        0         0  4528176   2103968  5893291  14795694   190656     61440   
1   240168    276281   354824    428279   481564    680828   521847    908028   
2   306787   1030254   146585    303986   245368    786556   222267    494216   
3    54156    334867    87905    642177    40929    279956    64040    455340   
4     5206     24618        0         0  1972980   3748940        0         0   

   2015_KG  2015_USD  2016_KG  2016_USD  2017_KG  2017_USD  2018_KG  2018_USD  \
0    47664     18240        0         0        0         0        0         0   
1   495428    741370   985739   1655417  2393468   4274650  3234168   5494321   
2   195896    524109   258072    687411   132688   1523699   169109    512519   
3    47609    222866   134106    499622    67594    266086    30835    126336   
4        0         0        0         0        0         0     6123     22631   

   2019_KG  2019_USD  2020_KG  2020_USD  2021_KG  2021_USD  2022_KG  2022_USD  \
0        0         0     1463      8550   181931    312926    66046    118618   
1  2419537   3826587  3299013   3869243  6522527   7192362  5076670   7156293   
2   209765    616274   300178    610793   111085    203554   220373    447893   
3   129852    376828   122253    363000    61884    264116   105395    404647   
4     3540      1353       28       126        0         0        0         0   

   Total_KG  Total_USD     Media_KG    Media_USD  Media_valor_litro  
0  39029799   25504484 4,336,644.33 2,833,831.56               0.65  
1  29214770   38719031 1,947,651.33 2,581,268.73               1.33  
2   3563355    9684567   237,557.00   645,637.80               2.72  
3   2509458    4746525   167,297.20   316,435.00               1.89  
4   1993000    3808552   284,714.29   544,078.86               1.91

In [125]:
df_inicial.shape

(73, 31)

In [130]:
df_melted = pd.melt(df_inicial, id_vars=['Pais'], var_name='Ano e Valor', value_name='Quantidade')
df_melted[['Ano', 'Valor']] = df_melted['Ano e Valor'].str.split('_', expand=True)
df_exportacao = df_melted.pivot(index=['Pais', 'Ano'], columns='Valor', values='Quantidade').reset_index()
df_exportacao.columns.name = None 
df_exportacao = df_exportacao.rename(columns={'KG': 'Litros', 'Pais' : 'Destino'})
df_exportacao['Origem'] = 'Brasil'
df_exportacao['Ano'] = pd.to_datetime(df_exportacao['Ano'], format='%Y')
df_exportacao['Litros'] = df_exportacao['Litros'].astype(float)
df_exportacao['USD'] = df_exportacao['USD'].astype(float)
df_exportacao.head()

Destino        Ano     Litros        USD  Origem
0  Alemanha, República Democrática 2008-01-01 265,742.00 429,970.00  Brasil
1  Alemanha, República Democrática 2009-01-01 225,086.00 393,482.00  Brasil
2  Alemanha, República Democrática 2010-01-01  27,715.00 138,666.00  Brasil
3  Alemanha, República Democrática 2011-01-01  36,070.00 144,150.00  Brasil
4  Alemanha, República Democrática 2012-01-01   8,189.00  56,342.00  Brasil

In [131]:
df_exportacao.shape

(1095, 5)

In [132]:
df_exportacao.describe()

Ano        Litros           USD
count                 1095      1,095.00      1,095.00
mean   2015-01-01 03:12:00     80,322.58    102,736.23
min    2008-01-01 00:00:00          0.00          0.00
25%    2011-01-01 00:00:00          0.00          0.00
50%    2015-01-01 00:00:00        450.00      1,375.00
75%    2019-01-01 00:00:00      7,189.50     23,645.50
max    2022-01-01 00:00:00 21,912,914.00 14,795,694.00
std                    NaN    791,339.02    657,169.62

In [133]:
df_exportacao.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1095 entries, 0 to 1094
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   Destino  1095 non-null   object        
 1   Ano      1095 non-null   datetime64[ns]
 2   Litros   1095 non-null   float64       
 3   USD      1095 non-null   float64       
 4   Origem   1095 non-null   object        
dtypes: datetime64[ns](1), float64(2), object(2)
memory usage: 42.9+ KB


In [134]:
df_exportacao['Destino'].nunique()

73

#### Anotação: Nos últimos 15 anos nós exportamos vinho para 73 Países.

# Gerando a Tabela Solicitada Pelo Head de Dados

In [139]:
df_total_por_pais = df_exportacao.groupby(['Origem','Destino']).sum(['Litros','USD']).sort_values('Litros', ascending=False)
df_total_por_pais.reset_index(inplace=True)
df_total_por_pais.head()

Origem         Destino        Litros           USD
0  Brasil          Rússia 39,029,799.00 25,504,484.00
1  Brasil        Paraguai 29,214,770.00 38,719,031.00
2  Brasil  Estados Unidos  3,563,355.00  9,684,567.00
3  Brasil           China  2,509,458.00  4,746,525.00
4  Brasil         Espanha  1,993,000.00  3,808,552.00

In [122]:
df_total_por_pais.to_excel('./ExportacaoVinhosTotais.xlsx', index=False)

# Coleta de informações com os dados que temos até o momento

In [143]:
df_total_por_pais['USD_por_Litro'] = df_total_por_pais['USD'] / df_total_por_pais['Litros']
df_total_por_pais.head(10)


# print("Top 5 países que mais importaram vinho do Brasil em Litros:")
# print(top5_litros)
# print("Top 5 países que mais importaram vinho do Brasil em USD:")
# print(top5_usd)
# print("País que menos importou vinho do Brasil em Litros:")
# print(min_litros)
# print("Média de Litros importados:")
# print(media_litros)
# print("Mediana de Litros importados:")
# print(mediana_litros)
# print("Média de USD importados:")
# print(media_usd)
# print("Mediana de USD importados:")
# print(mediana_usd)
# print("Preço médio por Litro:")
# print(preco_medio_por_litro.head())


Origem                          Destino        Litros           USD  \
0  Brasil                           Rússia 39,029,799.00 25,504,484.00   
1  Brasil                         Paraguai 29,214,770.00 38,719,031.00   
2  Brasil                   Estados Unidos  3,563,355.00  9,684,567.00   
3  Brasil                            China  2,509,458.00  4,746,525.00   
4  Brasil                          Espanha  1,993,000.00  3,808,552.00   
5  Brasil                            Haiti  1,791,603.00  2,327,208.00   
6  Brasil                      Reino Unido  1,239,551.00  4,711,464.00   
7  Brasil                    Países Baixos  1,236,154.00  3,791,611.00   
8  Brasil                            Japão  1,181,692.00  2,377,716.00   
9  Brasil  Alemanha, República Democrática    909,051.00  2,546,394.00   

   USD_por_Litro  
0           0.65  
1           1.33  
2           2.72  
3           1.89  
4           1.91  
5           1.30  
6           3.80  
7           3.07  
8           2.01  
9           2.80

In [141]:
top10_litros = df_total_por_pais.sort_values(by="Litros", ascending=False).head(10)[['Destino', 'Litros']]
top10_litros


Destino        Litros
0                           Rússia 39,029,799.00
1                         Paraguai 29,214,770.00
2                   Estados Unidos  3,563,355.00
3                            China  2,509,458.00
4                          Espanha  1,993,000.00
5                            Haiti  1,791,603.00
6                      Reino Unido  1,239,551.00
7                    Países Baixos  1,236,154.00
8                            Japão  1,181,692.00
9  Alemanha, República Democrática    909,051.00

In [142]:
top10_usd = df_total_por_pais.sort_values(by="USD", ascending=False).head(10)[['Destino', 'USD']]
top10_usd

Destino           USD
1                         Paraguai 38,719,031.00
0                           Rússia 25,504,484.00
2                   Estados Unidos  9,684,567.00
3                            China  4,746,525.00
6                      Reino Unido  4,711,464.00
4                          Espanha  3,808,552.00
7                    Países Baixos  3,791,611.00
9  Alemanha, República Democrática  2,546,394.00
8                            Japão  2,377,716.00
5                            Haiti  2,327,208.00

## Algumas conclusões:

1. **Top 10 países importadores em litros**: 
   - Rússia é o maior importador de vinho brasileiro em termos de volume, seguida por Paraguai, Estados Unidos, China, Espanha, Haiti, Reino Unido, Países Baixos, Japão e Alemanha.


2. **Top 10 países importadores em USD**: 
   - Paraguai é o maior importador em termos de valor, gastando mais de 38 milhões de dólares, mesmo tendo um volume menor em comparação com a Rússia. Isso indica que o vinho exportado para o Paraguai pode ser de um segmento mais caro ou de maior qualidade. 


3. **Preço médio por litro**: 
   - O preço por litro dá uma ideia do segmento ou qualidade do vinho exportado para diferentes países. 
   - Reino Unido, com 3.80 USD por litro, é o que paga o preço mais alto entre os top 10, o que sugere uma preferência por vinhos de maior qualidade ou segmento mais alto.
   - Rússia paga o menor preço por litro, 0.65 USD, o que sugere que eles podem estar focados em vinhos de segmento mais baixo ou podem estar aproveitando descontos por comprar em grande volume.

**Conclusão**: A análise dos dados revela diversas nuances no mercado de exportação de vinhos do Brasil. Por exemplo, enquanto a Rússia é o maior importador em volume, o Paraguai lidera em termos de valor, indicando diferentes preferências e estratégias de mercado entre os países. Além disso, o preço médio por litro oferece insights sobre a qualidade ou segmento do vinho que é mais popular em diferentes nações.

# Iniciando a Análise anualizada

In [145]:
df_exportacao.head()

Destino        Ano     Litros        USD  Origem
0  Alemanha, República Democrática 2008-01-01 265,742.00 429,970.00  Brasil
1  Alemanha, República Democrática 2009-01-01 225,086.00 393,482.00  Brasil
2  Alemanha, República Democrática 2010-01-01  27,715.00 138,666.00  Brasil
3  Alemanha, República Democrática 2011-01-01  36,070.00 144,150.00  Brasil
4  Alemanha, República Democrática 2012-01-01   8,189.00  56,342.00  Brasil

In [147]:
df_exportacao_anualizado = df_exportacao.copy()
df_exportacao_anualizado['USD_Litro'] = np.where(df_exportacao_anualizado['Litros'] > 0, df_exportacao_anualizado['USD'] / df_exportacao_anualizado['Litros'], 0)
df_exportacao_anualizado.sort_values(by=['Ano']).head(10)


Destino        Ano     Litros        USD  Origem  \
0    Alemanha, República Democrática 2008-01-01 265,742.00 429,970.00  Brasil   
930                         Suriname 2008-01-01     900.00   1,625.00  Brasil   
630                          Libéria 2008-01-01       0.00       0.00  Brasil   
465                             Gana 2008-01-01  18,168.00  25,642.00  Brasil   
165                       Cabo Verde 2008-01-01       0.00       0.00  Brasil   
360                          Equador 2008-01-01       0.00       0.00  Brasil   
735                   Nova Caledônia 2008-01-01       0.00       0.00  Brasil   
615                            Japão 2008-01-01 232,293.00 178,333.00  Brasil   
345           Emirados Arabes Unidos 2008-01-01   2,437.00  12,298.00  Brasil   
60                         Argentina 2008-01-01       0.00       0.00  Brasil   

     USD_Litro  
0         1.62  
930       1.81  
630       0.00  
465       1.41  
165       0.00  
360       0.00  
735       0.00  
615       0.77  
345       5.05  
60        0.00

In [151]:
import plotly.express as px
from scipy.stats import linregress

# Identificar todos os países únicos no dataframe
paises_unicos = df_exportacao_anualizado['Destino'].unique()

paises_alta = []

# Verificar a tendência de cada país
for pais in paises_unicos:
    df_pais = df_exportacao_anualizado[df_exportacao_anualizado['Destino'] == pais]
    if len(df_pais) > 1:
        slope, _, _, _, _ = linregress(df_pais['Ano'].astype('str').str[:4].astype(int), df_pais['Litros'])
        if slope > 10:
            paises_alta.append(pais)

# Para cada país com tendência de alta, criar um gráfico de barras
for pais in paises_alta:
    df_pais = df_exportacao_anualizado[df_exportacao_anualizado['Destino'] == pais]
    fig = px.bar(df_pais, x='Ano', y='Litros', title=f'Milhões Litros Importados para {pais} por Ano')
    fig.show()


In [ ]:
# df_total_por_pais.to_excel('./ExportacaoVinhosTotais.xlsx', index=False)

In [ ]:
# # import the library
# import dtale

# # open the analysis in browser
# dtale.show(df_exportacao).open_browser()